src/Mvc/Mvc.Core/test/ModelBinding/Metadata/DefaultModelMetadataTest.cs

```diff
@@ -339,6 +339,56 @@ public void IsBindingRequired_ReturnsFalse_ForTypes(Type modelType)
         Assert.False(isBindingRequired);
     }
 
+    [Fact]
+    public void IsParseableType_ReturnsTrue_WhenTryParseAllowed()
+    {
+        // Arrange
+        var provider = new EmptyModelMetadataProvider();
+        var detailsProvider = new EmptyCompositeMetadataDetailsProvider();
+
+        var key = ModelMetadataIdentity.ForType(typeof(int));
+        var cache = new DefaultMetadataDetails(key, new ModelAttributes(new object[0], null, null))
+        {
+            BindingMetadata = new BindingMetadata()
+            {
+                IsBindingFromTryParseAllowed = true,
+            },
+        };
+
+        var metadata = new DefaultModelMetadata(provider, detailsProvider, cache);
+
+        // Act
+        var isParseableType = metadata.IsParseableType;
+
+        // Assert
+        Assert.True(isParseableType);
+    }
+
+    [Fact]
+    public void IsParseableType_ReturnsFalse_WhenTryParseNotAllowed()
+    {
+        // Arrange
+        var provider = new EmptyModelMetadataProvider();
+        var detailsProvider = new EmptyCompositeMetadataDetailsProvider();
+
+        var key = ModelMetadataIdentity.ForType(typeof(string));
```

Comment:
This should be `int` (since it has a TryParse method)
There should also be a test that a type (like `string`) has a false `IsParseableType` value when `IsBindingFromTryParseAllowed` is true.

Reply:
I changed the tests.